src/Microsoft.DotNet.Cli.Utils/FileAccessRetryer.cs

```diff
@@ -47,5 +48,34 @@ public static class FileAccessRetrier
                 }
             }
         }
+
+        /// <summary>
+        /// Run Directory.Move and File.Move in Windows has a chance to get IOException with HResult 0x80070005. But this error is transient.
+        /// </summary>
+        internal static void RetryOnMoveAccessFailure(Action action)
+        {
+            const int ERROR_HRESULT_ACCESS_DENIED = unchecked((int)0x80070005);
+            int nextWaitTime = 1000;
+            int remainRetry = 3;
+
+            while (true)
+            {
+                try
+                {
+                    action();
+                    break;
+                }
+                catch (IOException e) when (e.HResult == ERROR_HRESULT_ACCESS_DENIED)
+                {
+                    Thread.Sleep(nextWaitTime);
+                    nextWaitTime *= 10;
```

Comment:
Maybe a factor of 5 (thus, 1 -> 5 -> 25)? 100 seconds for the final attempt seems like it will likely complete (I'd certainly ctrl-c long before then).